### PR TITLE
Improve test dataproviders

### PR DESCRIPTION
**Pull request by wernerkrauss**
As a result of my talk at StripeConEU I cleaned up SapphireTestTest a bit more.

### CODE REVIEW

File: tests/php/Control/SimpleResourceURLGeneratorTest.php [hunk 1]
```diff
@@ -64,4 +64,12 @@ public function testModuleResource()
             $generator->urlForResource($module->getResource('client/style.css'))
         );
     }
+
+    public function testURLForResourceFailsGracefully()
+    {
+        /** @var SimpleResourceURLGenerator $generator */
+        $generator = Injector::inst()->get(ResourceURLGenerator::class);
+        $url = @$generator->urlForResource('doesnotexist.jpg');
```
Comment (dhensby) [inline tests/php/Control/SimpleResourceURLGeneratorTest.php]: You probably want to assert a notice is triggered, not silence it like this.

Comment (wernerkrauss) [reply tests/php/Control/SimpleResourceURLGeneratorTest.php]: Probably, but that must be in another commit not by me... Did I PR against the correct branch?

Comment (dhensby) [reply tests/php/Control/SimpleResourceURLGeneratorTest.php]: Ah, ok. You'll need to rebase your work on the right branch

**Comment by robbieaverill** [comment]
Rebased

File: tests/php/Dev/SapphireTestTest.php [hunk 2]
```diff
@@ -71,11 +90,26 @@ public function testCreateMemberWithPermission()
...
         $list = $this->generateArrayListFromItems($itemsForList);
 
-        $this->assertListAllMatch($match, $list);
+        $this->assertListAllMatch($match, $list, $message);
+    }
+
+    /**
+     * generate SS_List as this is not possible in dataProvider
+     *
+     * @param $itemsForList array
```
Comment (robbieaverill) [inline tests/php/Dev/SapphireTestTest.php]: This should be `@param array $itemsForList`

File: tests/php/Dev/SapphireTestTest/DataProvider.php [hunk 3]
```diff
@@ -15,14 +15,19 @@ class DataProvider implements TestOnly
...
     /**
      * @return array
      */
     public static function provideEqualListsWithEmptyList()
     {
         return array_merge(
-            [ //empty list
-                [
+            [
+                'emtpyLists' => [
```
Comment (robbieaverill) [inline tests/php/Dev/SapphireTestTest/DataProvider.php]: Typo

File: tests/php/Dev/SapphireTestTest/DataProvider.php [hunk 4]
```diff
@@ -124,28 +129,28 @@ public static function provideNonEqualLists()
     public static function provideNotContainingList()
     {
         return [
-            [ //empty list
+            'listIsEmtpy' => [
```
Comment (robbieaverill) [inline tests/php/Dev/SapphireTestTest/DataProvider.php]: Typo

File: tests/php/Dev/SapphireTestTest/DataProvider.php [hunk 5]
```diff
@@ -159,14 +164,14 @@ public static function provideNotContainingList()
...
-        $list = [
-            ['FirstName' => 'Ingo', 'Surname' => 'Schommer', 'Locale' => 'en_US'],
-            ['FirstName' => 'Sam', 'Surname' => 'Minnee', 'Locale' => 'en_US']
-        ];
-
         return [
-            [[], $list], //empty match
-            [['Locale' => 'en_US'], $list] //all items have this field set
+            'emptyMatch' => [[], self::$memberList,
+            'empty list did not match'],
```
Comment (robbieaverill) [inline tests/php/Dev/SapphireTestTest/DataProvider.php]: I think this needs a quick reformat to pass PSR-2 linting =)

**Comment by dhensby** [comment]
@robbieaverill I've addressed your comments and pushed up

**Pull request merged**